### PR TITLE
some pandas things

### PR DESCRIPTION
**Pull request by hamima-halim**
* each thread only needs a small chunk of GTFS, since we know a priori which routes to looks at. smaller dfs, faster merges. 
* when calculating headways, we're assuming that we're working over batches--not true for gobble. filter where we can. i've kept the batch code in case we ever want to use it.
* sample files and tests!

### CODE REVIEW

File: src/tests/test_gtfs.py
Comment (devinmatte) [inline src/tests/test_gtfs.py:1]: 😍 Tests!

**Comment by hamima-halim** [comment]
https://github.com/transitmatters/gobble/pull/55/files#diff-fc6841882a7f8a0238936d3b5611a6027ba6345cde8dd48be25a175d317f8ecf ian's diff is better, ignore mine--will just keep the test probs

File: src/gtfs.py [hunk 1]
```diff
@@ -209,3 +200,80 @@ def add_gtfs_headways(events_df: pd.DataFrame, all_trips: pd.DataFrame, all_stop
...
+    with merge_asof.
+    https://pandas.pydata.org/docs/reference/api/pandas.merge_asof.html
+    This function is ADAPTED from historical bus headway calculations
+    https://github.com/transitmatters/t-performance-dash/blob/ebecaca071b39d8140296545f2e5b287915bc60d/server/bus/gtfs_archive.py#L90
+
+    NB 1: event times are converted to pd timestamps in this fuction for pandas merge manipulation,
+    but will be converted back into datetime.datetime for serialization purposes. careful!
+    NB 2: ensure that all times are in US/Eastern!
+    """
+    # TODO: I think we need to worry about 114/116/117 headways?
```
Comment (devinmatte) [inline src/gtfs.py:220]: We're also adding more routes like this like `220/221/222`

**Review by devinmatte — APPROVED** [review]
It seems good to me, but it is a big change, so we'll just need to watch the traces

**Comment by devinmatte** [comment]
Ah speaking of tests, the tests don't seem to pass on CI (also I should split lint and tests into two different CI jobs)

**Pull request merged**